lib/cli/src/commands/app/mod.rs

```diff
@@ -130,20 +130,6 @@ pub async fn deploy_app_verbose(
 
     eprintln!("Deploying app {pretty_name}...\n");
 
-    let (owner, app_opt) = if let Some(owner) = owner {
```

Comment:
I think there was a reason for this pre-fetch, because otherwise there could be some odd behaviour when the owner didn't match the app id.
I don't remember the exact details.
Need to make sure the backend does the right thing without this.

What happens if the specified app_id does not match a manually specified `--owner`?

Reply:
the backend has validation for that, so we should be good.

Reply:
```
╰─▶ 2: GraphQL API failure: Expected the app specified by app_id (`app_id`) to be `test-tiinyhost`, but found 
`ayush-test-tiinyhost` instead. Check your `name` field in the config to make sure it's correct.
```